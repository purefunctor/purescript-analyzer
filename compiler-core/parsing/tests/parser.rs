use test_each_file::test_each_file;

test_each_file! { in "./compiler-core/parsing/tests/parser" => |content: &str| {
    let lexed = lexing::lex(content);
    let tokens = lexing::layout(&lexed);
    let (parsed, errors) = parsing::parse(&lexed, &tokens);
    let node = parsed.syntax_node();
    insta::assert_debug_snapshot!((node, errors));
}}

test_each_file! { in "./compiler-core/parsing/tests/parser" as lossless => |content: &str| {
    let lexed = lexing::lex(content);
    let tokens = lexing::layout(&lexed);
    let (parsed, _) = parsing::parse(&lexed, &tokens);
    let node = parsed.syntax_node();
    assert_eq!(node.to_string(), content);
}}

test_each_file! { in "./compiler-core/parsing/tests/parser" as stability => |content: &str| {
    let lexed = lexing::lex(content);
    for index in 0..lexed.len() - 1 {
        let partial = lexed.text_in_range(0..index + 1);
        let lexed = lexing::lex(partial);
        let tokens = lexing::layout(&lexed);
        let (parsed, _) = parsing::parse(&lexed, &tokens);
        let node = parsed.syntax_node();
        assert_eq!(node.to_string(), partial);
    }
}}
