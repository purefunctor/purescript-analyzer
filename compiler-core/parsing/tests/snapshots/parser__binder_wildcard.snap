---
source: compiler-core/parsing/tests/parser.rs
expression: "(node, errors)"
snapshot_kind: text
---
(
    Module@0..44
      ModuleHeader@0..27
        MODULE@0..6 "module"
        ModuleName@6..21
          Annotation@6..7
            TEXT@6..7 " "
          UPPER@7..21 "BinderWildcard"
        Annotation@21..22
          TEXT@21..22 " "
        WHERE@22..27 "where"
      LAYOUT_START@27..27 ""
      ModuleImports@27..27
      ModuleStatements@27..43
        ValueEquation@27..43
          Annotation@27..29
            TEXT@27..29 "\n\n"
          LOWER@29..37 "wildcard"
          FunctionBinders@37..39
            BinderWildcard@37..39
              Annotation@37..38
                TEXT@37..38 " "
              UNDERSCORE@38..39 "_"
          Unconditional@39..43
            Annotation@39..40
              TEXT@39..40 " "
            EQUAL@40..41 "="
            WhereExpression@41..43
              ExpressionInteger@41..43
                Annotation@41..42
                  TEXT@41..42 " "
                INTEGER@42..43 "0"
      LAYOUT_END@43..43 ""
      Annotation@43..44
        TEXT@43..44 "\n"
      END_OF_FILE@44..44 ""
    ,
    [],
)
